File: SOLID/SRP/BankSystem.java
class Teller {
    // Responsibility: Handle physical cash deposits
    public void depositCash(double amount, BankLedger ledger) {
        System.out.println("Teller: Safely counting $" + amount + "...");
        ledger.record("Cash Deposit: $" + amount);
    }
}

class LoanOfficer {
    // Responsibility: Handle loan evaluation
    public boolean evaluateMortgage(int creditScore, double income) {
        System.out.println("Loan Dept: Running credit risk algorithms...");
        return creditScore > 700 && income > 50000;
    }
}

class BankLedger {
    // Responsibility: Store ledger entries
    public void record(String message) {
        System.out.println("Ledger: Entry saved -> " + message);
    }
}

public class BankSystem {
    public static void main(String[] args) {
        BankLedger ledger = new BankLedger();
        Teller teller = new Teller();
        LoanOfficer loanOfficer = new LoanOfficer();

        teller.depositCash(500, ledger);

        boolean result = loanOfficer.evaluateMortgage(750, 60000);
        System.out.println("Mortgage Result: " + (result ? "Approved" : "Denied"));
    }
}

File: SOLID/SRP/BankSystem.py
class Teller:
    # Responsibility: Handle cash deposits
    def deposit_cash(self, amount, ledger):
        print(f"Teller: Safely counting ${amount}...")
        ledger.record(f"Cash Deposit: ${amount}")


class LoanOfficer:
    # Responsibility: Evaluate mortgage applications
    def evaluate_mortgage(self, credit_score, income):
        print("Loan Dept: Running credit risk algorithms...")
        return credit_score > 700 and income > 50000


class BankLedger:
    # Responsibility: Record ledger entries
    def __init__(self):
        self.entries = []

    def record(self, message):
        self.entries.append({"time": "now", "action": message})
        print("Ledger: Entry saved successfully.")


def main():
    ledger = BankLedger()
    teller = Teller()
    loan_officer = LoanOfficer()

    teller.deposit_cash(500, ledger)

    approved = loan_officer.evaluate_mortgage(750, 60000)
    print(f"Mortgage Result: {'Approved' if approved else 'Denied'}")


main()
